### DISCUSSION TITLE
Minio Object versioning storage utilization

### DISCUSSION THREAD

**Discussion by allahshukur-ahmadzada** [minio/minio#15787] [Q&A] [answered]
Hello, can anyone briefly explain how storage needs affected by versioning?

**Comment by klauspost** [comment, marked as answer]
Versions are stored independently, but with shared metadata. Not sure what other information you are looking for.

**Comment by allahshukur-ahmadzada** [comment]
So 3 versions means almost 3x storage utilization?

**Reply by klauspost** [reply]
Yes.

**Reply by allahshukur-ahmadzada** [reply]
Thank you so much, one more question I can`t see option to define retention period by number of versions, like I would like to keep last 3 items. Possible?

**Reply by klauspost** [reply]
`mc ilm add` has the following option:

```
  --newer-noncurrentversions-transition value  the number of noncurrent versions to retain. If there are this many more recent noncurrent versions they wibe transitioned (default: 0)
```

**Reply by allahshukur-ahmadzada** [reply]
thank you

**Reply by allahshukur-ahmadzada** [reply]
I wonder why docs does not have that option?
https://min.io/docs/minio/linux/administration/object-management/object-lifecycle-management.html

**Reply by klauspost** [reply]
It is fairly new.

**Reply by ravindk89** [reply]
We just finished a large-scale migration of content. Now were playing release catchup. You can see the backlog at github.com/minio/docs under our issues :)

We'll get to it @allahshukur-ahmadzada , along with a larger clean up of our ILM docs.
